test/integration/redirects_test.rb

```diff
@@ -142,4 +162,197 @@ def test_show_past_glossary_term
+  # request_to_be_curator (post)  Herbaria::CuratorRequest#create (post)
+  # show_herbarium (get)          show (get)
+  # show_herbarium (post)         Herbaria::Curators#create (post)
+
+  def test_create_herbarium_get
+    login(rolf)
+    assert_old_url_redirects_to_new_path(
+      :get, "/herbarium/create_herbarium", new_herbarium_path
+    )
+  end
+
+  def test_create_herbarium_post
+    login(rolf)
+    assert_old_url_redirects_to_new_path(
+      :post, "/herbarium/create_herbarium", new_herbarium_path
+    )
+  end
+
+  def test_delete_herbarium_curator_post
+    nybg = herbaria(:nybg_herbarium)
+    # make sure nobody messed up the fixtures
+    assert(nybg.curator?(rolf))
+    assert(nybg.curator?(roy))
+
+    login(rolf)
+
+    # Test the results of the redirect because
+    # There is no way to test the redirect directly (unlike other actions).
+    # Due to the routing scheme, Rails actually follows the redirect.
+    assert_old_url_redirects_to_new_path(
+      :post,
+      "/herbarium/delete_curator/#{nybg.id}?user=#{roy.id}",
+      herbarium_path(nybg)
+    )
+    assert_response(:success)
+    assert_not(nybg.reload.curator?(roy))
+  end
+
+  def test_destroy_herbarium
+    login(rolf)
+    assert_old_url_redirects_to_new_path(
+      :post,
+      "/herbarium/destroy_herbarium/#{herbaria(:rolf_herbarium).id}",
+      herbarium_path(herbaria(:rolf_herbarium))
+    )
+  end
+
+  def test_edit_herbarium_get
+    login(rolf)
+    assert_old_url_redirects_to_new_path(
+      :get,
+      "/herbarium/edit_herbarium/#{herbaria(:rolf_herbarium).id}",
+      edit_herbarium_path(herbaria(:rolf_herbarium))
+    )
+  end
+
+  def test_edit_herbarium_post
+    login(rolf)
+    assert_old_url_redirects_to_new_path(
+      :post,
+      "/herbarium/edit_herbarium/#{herbaria(:rolf_herbarium).id}",
+      edit_herbarium_path(herbaria(:rolf_herbarium))
+    )
+  end
+
+  def test_herbarium_search
+    assert_old_url_redirects_to_new_path(
+      :get, "/herbarium/herbarium_search", herbaria_searches_path
+    )
+  end
+
+  def test_herbarium
+    assert_old_url_redirects_to_new_path(
+      :get, "/herbarium", herbaria_nonpersonals_path
+    )
+  end
+
+  def test_index_herbarium
+    assert_old_url_redirects_to_new_path(
+      :get, "/herbarium/index", herbaria_path
+    )
+  end
+
+  def test_list_herbaria
+    assert_old_url_redirects_to_new_path(
+      :get, "/herbarium/list_herbaria", herbaria_alls_path
+    )
+  end
+
+  # merge_herbaria (get)          Herbaria::Merges#new (get)
+  def test_merge_herbaria
+    fundis = herbaria(:fundis_herbarium)
+    assert_true(fundis.owns_all_records?(mary))
+    marys = mary.create_personal_herbarium
+    login("mary")
+
+    get("/herbarium/merge_herbaria?this=#{fundis.id}&that=#{marys.id}")
+    # fundis ends up being the destination because it is older.
+    assert_equal(herbaria_path(id: fundis.id), @response.request.fullpath)
```

Comment:
I don't understand why the response is :success rather than a 301. 
Obviously a consequence of my routing. 
How can I fix this?

Reply:
Should the action at Herbaria::Merges#new be rather Herbaria::Merges#create, and then instead of `get("/herbarium/merge_herbaria?this=#{fundis.id}&that=#{marys.id}")
` you'd do a `post` to `merge_herbaria`?

Reply:
Good point. 
I was (and still am) confused about what should be the right action.
Under some circumstances it actually performs the merger, which is both an update of one herbarium and a destroy of the other. In other cases, it doesn't change any herbarium, but merely sends an email to the webmaster.